Update key_vault.py to use User-Managed-Identity

Tom-Ganor opened the pull request:
This change will support using MSI with User-Assigned-Managed-Identity (a client_id is needed to be provided as a parameter).

Tom-Ganor committed:
Update key_vault.py

This change will support using MSI with User-Assigned-Managed-Identity (a client_id is needed to be provided as a parameter).

dany74q commented on keyvaultlib/key_vault.py:
```diff
@@ -43,7 +43,7 @@ def __init__(self, client_id=None, client_secret=None, tenant_id=None, use_msi=F
 
         if use_msi:
-            msi_creds = MSICredentials(resource=self.KEY_VAULT_RESOURCE_URL)
+            msi_creds = MSICredentials(resource=self.KEY_VAULT_RESOURCE_URL, client_id=client_id)
```
Please add relevant documentation to the client_id param above, also - verify this does not break old MSICredentials version that did not accept a client_id parameter, if there were any.

Thanks !

Tom-Ganor replied:
@dany74q  Done.

Tom-Ganor committed:
Update key_vault.py

Adding proper client_id documentation + verified this does not break old MSICredentials versions.
This change will break only versions of msrestazure that are < 0.4.19 (this package's requirement is version 0.5.0).
